framework/src/samplers/Sampler.C

```diff
@@ -352,64 +369,48 @@ Sampler::getRandl(unsigned int index, uint32_t lower, uint32_t upper)
 dof_id_type
 Sampler::getNumberOfRows() const
 {
+  checkReinitStatus();
   return _n_rows;
 }
 
 dof_id_type
 Sampler::getNumberOfCols() const
 {
+  checkReinitStatus();
   return _n_cols;
 }
 
 dof_id_type
 Sampler::getNumberOfLocalRows() const
 {
+  checkReinitStatus();
   return _n_local_rows;
 }
 
 dof_id_type
 Sampler::getLocalRowBegin() const
 {
+  checkReinitStatus();
   return _local_row_begin;
 }
 
 dof_id_type
 Sampler::getLocalRowEnd() const
 {
+  checkReinitStatus();
   return _local_row_end;
 }
 
-// DEPRECATED: Everything below should removed when apps are updated to new syntax
-std::vector<DenseMatrix<Real>>
-Sampler::sample()
-{
-  return std::vector<DenseMatrix<Real>>();
-}
-
-std::vector<DenseMatrix<Real>>
-Sampler::getSamples()
-{
-  mooseDoOnce(mooseDeprecated(
-      "getSamples is being removed, use getNextLocalRow, getLocalSamples, or getGlobalSamples."));
-
-  _generator.restoreState();
-  sampleSetUp();
-  std::vector<DenseMatrix<Real>> output = sample();
-  sampleTearDown();
-
-  mooseAssert(output.size() > 0,
-              "It is not acceptable to return an empty vector of sample matrices.");
-
-  return output;
-}
-
-// TODO: Remove this and restore to pure virtual when deprecated syntax is removed
-Real Sampler::computeSample(dof_id_type, dof_id_type) { return 0.0; }
-
-double
-Sampler::rand(const unsigned int index)
+void
+Sampler::checkReinitStatus() const
 {
-  mooseDoOnce(mooseDeprecated("rand(() is being removed, use getRand()"));
-  mooseAssert(index < _generator.size(), "The seed number index does not exists.");
-  return _generator.rand(index);
+  if (_needs_reinit)
+    mooseError("A call to 'setNumberOfRows()/Columns()' was made after initialization, as such the "
+               "expected Sampler output has changed and a new sample must be created. However, a "
+               "call to Sampler::reinit() was not performed. The renit() method is automatically "
+               "called during Sampler execution, which occurs according to the 'execute_on' "
+               "settings of the Sampler object. An adjustment to this parameter may be required. "
+               "It is recommended that calls to 'setNumberOfRows()/Columns() occur within the "
+               "Sampler::executeSetUp() method; this will ensure that the reinitialize is handled "
+               "correctly. Have a nice day.");
```

Comment:
I like it. Sounds sassy.

Reply:
https://www.youtube.com/watch?v=3UunQXXxAWY